Disable Watson TTS Telemetry

poofyteddy opened the pull request:
## Description:

Will disable telemetry by default for everyone. But i don't see this as a bad thing, maybe i'm wrong

As in the doc https://cloud.ibm.com/apidocs/text-to-speech?code=python#data-collection
I think leaving telemetry enable break the idea of privacy a self hosted solution like HA allow. So this option felt necessary.

Can't test it right now because HA is down for unrelated reason
First time i do any edition to HA, so i surely missed stuff, plus i am really not a dev ...

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here> Not needed, no change to the front end

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.
  - [ ] I have followed the [development checklist][dev-checklist]

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

poofyteddy committed:
Disable telemetry by default

As in the doc https://cloud.ibm.com/apidocs/text-to-speech?code=python#data-collection
Still need to understand how config work in ha

poofyteddy committed:
enable the use of enable_telemetry in conf file

This should do it from what i understand

amelchio commented on homeassistant/components/watson_tts/tts.py:
```diff
@@ -99,6 +102,7 @@ def get_engine(hass, config):
     supported_languages = list({s[:5] for s in SUPPORTED_VOICES})
     default_voice = config[CONF_VOICE]
     output_format = config[CONF_OUTPUT_FORMAT]
+    service.set_default_headers({'x-watson-learning-opt-out': config[CONF_TELEMETRY]})
```
I think true/false is being swapped here.

poofyteddy replied:
I simply disabled it on my side with custom component, but i though i might add a check
and your are right, default should be true if disable by default ... i think i changed my mind about what should be default to many time XD

poofyteddy replied:
should be better

amelchio reviewed (dismissed):
Do we even need the option, why would anyone ever enable it?

poofyteddy committed:
Default to disable

amelchio commented:
It was actually better before. We prefer positively named options because double negatives like `disable_telemetry: false` get weird.

But no option is even better, so please answer my above question: Do we even need the option, why would anyone ever enable it?

poofyteddy commented:
Sorry @amelchio, Github is an unfriendly thing for me, and i was looking for a reply button ...
In the way i see thing, you should never remove an option. Giving the user a chose about it feel right.
99% of people won't actively look to enable it, but i think there is a 1% who wish to help development of the tech by sharing data

amelchio commented:
> In the way i see thing, you should never remove an option.

The Home Assistant view is that we should not try to please everyone because we will fail. Each option is extra complexity and overhead that should be avoided if almost nobody wants it anyway.

poofyteddy committed:
Switch to static disable

as requested by @amelchio, the need to enable telemetry to small to make thing more complicated.

poofyteddy commented:
Then it's way easier :) i rollback to just putting the config needed to always opt-out of telemetry.

amelchio commented:
> Then it's way easier :)

Exactly. And then we make less mistakes. Please remove the WIP label once you have tested this.

poofyteddy commented:
After testing it it work as expected, i receive my audio file.
i could not do a end to end test since my tts player (squeezebox) don't seam to be willing to play the file, but if i download it with the same url i can play it on my computer without issue.

amelchio commented:
We need to get all CI checks to pass in order to move this along but I am not sure why some failed?

amelchio committed:
Black
